pkg/resource/db_instance/hooks.go

```diff
@@ -194,3 +195,16 @@ func (rm *resourceManager) restoreDbInstanceFromDbSnapshot(
 	}
 	return &resource{r.ko}, nil
 }
+
+// RDS will choose preferred engine minor version if only
+// engine major version is provided and controler should not
+// treat them as different, sunch as spec has 14, status has 14.1
+// controller should treat them as same
+func reconcileEngineVersion(
+	a *resource,
+	b *resource,
+) {
+	if a != nil && b != nil && strings.HasPrefix(*b.ko.Spec.EngineVersion, *a.ko.Spec.EngineVersion) {
+		a.ko.Spec.EngineVersion = b.ko.Spec.EngineVersion
+	}
```

Comment:
I think this logic will fail if you set the minor version in the spec. For instance, if you set `Spec = 14.1` and `Status = 14.0`, this will not upgrade the cluster version. Is that expected?

Reply:
I think it's working. like `strings.HasPrefix("14.1", "14,0")` is returning false , so controller will not do the override here and treat them as different

spec file has 14.2 and then i modified it to 14.3 and it's getting the diff.
`1.6557526651883025e+09	INFO	ackrt	desired resource state has changed	{"account": "980401675890", "role": "", "region": "us-west-2", "kind": "DBInstance", "namespace": "default", "name": "brucegu-test-tag-3", "is_adopted": false, "generation": 11, "diff": [{"Path":{"Parts":["Spec","EngineVersion"]},"A":"14.3","B":"14.2"}]}`

Reply:
Oh oh I see. You're using `strings.HasPrefix`. Yep that makes sense